### PR TITLE
native: mark internal routines

### PR DESCRIPTION
**Pull request by kortschak**
@btracey Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
Fixes #35.

**Comment by btracey** [comment]
Is the policy that a function is internal by default? For example, I could see mat64 wanting to use dlatrs. If we decided that's true, we would just remove the internal comment?

**Comment by kortschak** [comment]
Yes, and we would add Dlatrs to the lapack64 interface.

I intially went through and started to make these unexported, but realised (late) that then testing would break.

**Comment by kortschak** [comment]
*ping*

**Comment by btracey** [comment]
clapack is internal by default probably.

If we really want the functions to be uncallable, we could have a wrapper type that lives in native_test.go and exports them. It's not clear we want this. On the one hand, it's nice to allow direct access for advanced users, but on the other hand I don't know that we want to promise API stability for them. Maybe stability only extends to lapack64? We already can't promise to keep clapack stable.

File: native/dgels.go [hunk 1]
```diff
@@ -33,11 +33,13 @@ import (
...
 // this submatrix is of size n×nrhs, and of size m×nrhs otherwise.
 //
-// Work is temporary storage, and lwork specifies the usable memory length.
+// work is temporary storage, and lwork specifies the usable memory length.
 // At minimum, lwork >= max(m,n) + max(m,n,nrhs), and this function will panic
 // otherwise. A longer work will enable blocked algorithms to be called.
 // In the special case that lwork == -1, work[0] will be set to the optimal working
 // length.
+//
+// Dgels is an internal routine. It is exported for testing purposes.
```
Comment (btracey) [inline native/dgels.go]: It's part of lapack.Float64, so I don't think this is internal. We don't actually use it anymore in mat64, so we could remove it if we would like.

**Comment by btracey** [comment]
LGTM with Dgels fixed.

File: native/dgels.go (hunk 1 above)
Comment (kortschak) [inline native/dgels.go]: I missed that it was there. We need to make a decision about whether to keep it before the API promise.

**Comment by kortschak** [comment]
>Maybe stability only extends to lapack64?

Possibly.

>We already can't promise to keep clapack stable.

Yes.

**Pull request merged**